### PR TITLE
Move safe code out of unsafe blocks

### PR DESCRIPTION
**Pull request by gootorov**
Noticed there's some safe code which is put into unsafe blocks.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/gleam/146)

### CODE REVIEW

**Comment by nox** [comment]
This can have the opposite effect and make you focus only on the `unsafe` parts which may be unsound if the safe parts around it aren't actually implementing the invariants required by that `unsafe` block.

File: src/gl_fns.rs [hunk 1]
```diff
@@ -109,43 +109,43 @@ impl Gl for GlFns {
     }
 
     fn gen_buffers(&self, n: GLsizei) -> Vec<GLuint> {
+        let mut result: Vec<_> = repeat(0 as GLuint).take(n as usize).collect();
```
Comment (kvark) [inline src/gl_fns.rs]: better just use `vec![0; n as usize]`

Comment (gootorov) [reply src/gl_fns.rs]: Sure, added that.

Comment (gootorov) [reply src/gl_fns.rs]: By the way, what do you think about using `vec![0 as _; n as _]`, it will infer all types automatically and it's even less typing.

**Comment by kvark** [comment]
@nox  I looked at the changed and didn't have any concerns. If you feel that some of the stuff should be left in `unsafe` please provide an example that gives us an idea of the invariants the unsafe block may fail to meet.

File: src/gl_fns.rs (hunk 1 above)
Comment (kvark) [reply src/gl_fns.rs]: heheh, I don't think the first part is needed. `0` should be coerced without a cast. As for the second, I think it's fairly straightforward, so `_` is not confusing.

**Comment by nox** [comment]
@kvark Ack, feel free to land it whenever you want then!

**Comment by jdm** [comment]
@bors-servo r+
This looks like an improvement to me.

**Pull request merged**